Beta_Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -0,0 +1,1648 @@
+
+    if response.get('errors'):
+        return_error(response.get('errors'))
+    if 'data' in response:
+        return response.get('data')
+    return {}
+
+
+def create_query():
+
+    query = demisto.args().get('query')
+    from_date = demisto.args().get('from_date')
+    to_date = demisto.args().get('to_date')
+
+    query_id = create_query_request(query, from_date, to_date)
+
+    demisto.results('The query ID is ' + query_id)
+
+
+def get_events_request(query_id=None, limit=None):
+
+    endpoint_url = 'events'
+
+    params = {
+        'query_id': query_id,
+        'limit': limit
+    }
+
+    response = http_request('GET', endpoint_url, params)
+    if response.get('errors'):
+        return_error(response.get('errors'))
+    if 'data' in response:
+        return response.get('data')
+    return {}
+
+
+def get_events():
+    """
+    Get all Deep Visibility events from query
+    """
+    contents = []
+    context_entries = []
+    context = {}
+
+    query_id = demisto.args().get('query_id')
+    limit = int(demisto.args().get('limit', 10))
+
+    events = get_events_request(query_id, limit)
+    if events:
+        for event in events:
+            contents.append({
+                'Sha1': event.get('sha1'),
+                'DstIp': event.get('dstIp'),
+                'SrcIp': event.get('srcIp'),
+                'FileFullName': event.get('fileFullName'),
+                'ProcessName': event.get('processName')
+            })
+
+            context_entries.append({
+                'Sha1': event.get('sha1'),
+                'DstIp': event.get('dstIp'),
+                'SrcIp': event.get('srcIp'),
+                'FileFullName': event.get('fileFullName'),
+                'ProcessName': event.get('processName')
+            })
+
+        context['SentinelOne.Event(val.sha1 && val.sha1 === obj.sha1)'] = context_entries
+
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': contents,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('SentinelOne Events', contents, removeNull=True),
+        'EntryContext': context
+    })
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('time')
+
+    # handle first time fetch
+    if last_fetch is None:
+        last_fetch, _ = parse_date_range(FETCH_TIME, to_timestamp=True)
+
+    incidents = []
+    threats = get_threats_request()
+    for threat in threats:
+        # If no fetch threat rank is provided, bring everything, else only fetch above the threshold
+        if not FETCH_THREAT_RANK or (FETCH_THREAT_RANK and threat.get('rank') >= FETCH_THREAT_RANK):
+            incident = threat_to_incident(threat)
+            incident_date = date_to_timestamp(incident['occurred'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            # update last run
+            if incident_date > last_fetch:
+                last_fetch = incident_date
```

Comment:
we don't compare only last_fetch anymore, but take another var which is "max" time of events being fetched, then compare if current_fetch > max_fetch and update it accordingly and pass it to setLastRun. Can see it in any new integration with fetch, it is the same in most integrations.